### ISSUE TITLE
DatePicker do not render properly in Android

### ISSUE THREAD

**Issue by FrancoisM** [nventive/Uno#188]
## I'm submitting a...

Bug report (I searched for similar issues and did not find one)

## Current behavior

When adding a DatePicker on a page, it renders directly in its "edit" mode (i.e the "popup" to select a day with "cancel" and "ok" buttons at the bottom).
Moreover, that "popup" isn't one as it is constrained to its parent (say a row in a grid) instead of being in front of all other controls in the page. As such, if it doesn't have enough room, it is cropped and the user cannot see the "cancel" and "ok" buttons.

## Expected behavior
The DatePicker should display the bounded date and only show the "popup" once the user tap on the control. Then the "popup" should be centered in the in front of every other controls. 

## Minimal reproduction of the problem with instructions

The following xaml should be enough:

    <Grid>
        <Grid.RowDefinitions>
             <RowDefinition Height = "*">
             <RowDefinition Height = "*">
             <RowDefinition Height = "*">
             <RowDefinition Height = "*">
        </Grid.RowDefinitions>
        <DatePicker Date="{Binding Date}" Grid.Row=0 />
        <DatePicker Date="{Binding Date}" Grid.Row=1 />
        <DatePicker Date="{Binding Date}" Grid.Row=2 />
        <DatePicker Date="{Binding Date}" Grid.Row=3 />
    </Grid>

## Environment

```
Nuget Package: Uno.UI

Package Version(s): 1.31.0-dev.184

Affected platform(s):
- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [x] 2017 (version: 15.8.2 )
- [ ] 2017 Preview (version: )
- [ ] for Mac (version: )

```

**Comment by HaoLife** [comment]
![image](https://user-images.githubusercontent.com/25171162/46158908-81f33600-c2b1-11e8-8c57-c2688d2ed37b.png)
Display of DatePicker on Android

![image](https://user-images.githubusercontent.com/25171162/46158973-ae0eb700-c2b1-11e8-9ff1-8b2976fde6d8.png)
Display of DatePicker on Windows


Whether Android can display the same as Windows
@jeromelaban

**Comment by FrancoisM** [comment]
Uno gave me that [style ](https://gist.github.com/davidjohnoliver/692553bf9a66009c31cd23b7a7e2b8ec)that (almost) works.
When date is defined it almost looks like UWP (except no border) and we tap on it, it opens the android date selector fine.

**Comment by davidjohnoliver** [comment]
Yes the default DatePicker style needs to be fixed in Uno, the workaround is to base off the style linked by @FrancoisM .

**Comment by TopperDEL** [comment]
Is there a plan on when this is going to be merged? I desperately Need a working DatePicker on Android.
The given Workaround leads to Problems with UWP.

**Comment by TopperDEL** [comment]
It seems I have the same issues on iOs. Is there a working reference-implementation for Date- and TimePicker available where I can check my implementation?
